### PR TITLE
Percentage circle

### PR DESCRIPTION
**Pull request by hankeliu2015**
Added remaining circle and styles

### CODE REVIEW

File: src/popup/components/character-countdown.tsx [hunk 1]
```diff
@@ -2,34 +2,49 @@ import * as React from 'react'
...
-          {Math.abs(remaining)}
-        </text>
-      )}
-    </svg>
+    <div
+      className="character-countdown"
+      role="progressbar"
+      aria-valuemax={100}
+      aria-valuemin={0}
+      aria-valuenow={Math.round(percentageCompleted) >= 100 ? 100 : Math.round(percentageCompleted)}
```
Comment (will-weiss) [inline src/popup/components/character-countdown.tsx:24]: Looks good! We can trim this up even further by writing

```typescript
Math.min(Math.round(percentageCompleted), 100)
```

File: src/tests/integration/character-countdown.test.ts [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+  describe('when approaching the character limit', () => {
+    const almost280Chars = '@zing.com ' + 'x'.repeat(260)
+    before(() => {
+      mocks.backgroundWindow.store.dispatchers.updateEditorState({
+        editorState: fromText(almost280Chars),
+      })
+    })
+
+    it('renders a more full progress circle', () => {
+      // TODO: Hank to write this test
```
Comment (will-weiss) [inline src/tests/integration/character-countdown.test.ts:55]: Remove these TODOs

File: src/tests/integration/character-countdown.test.ts [hunk 3]
```diff
@@ -0,0 +1,154 @@
+import { expect } from 'chai'
+import { createMocks } from './mocks'
+import { runBackground } from './steps/run-background'
+import { mountPopup } from './steps/mount-popup'
+import { ensureActiveTab } from '../../selectors'
+import { fromText, getPlainText } from '../../draft-js-utils'
+
+// TODO: Hank to remove this .only when done writing these tests
+describe.only('character countdown', () => {
```
Comment (will-weiss) [inline src/tests/integration/character-countdown.test.ts:9]: Remove this .only

File: src/tests/integration/character-countdown.test.ts [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+    })
+
+    it('renders a more full progress circle', () => {
+      // TODO: Hank to write this test
+      const characterCountdown = mocks.app().querySelector('.character-countdown')! as HTMLDivElement
+      const progressCircle = characterCountdown.querySelector('circle.progress')! as SVGCircleElement
+      const remainingRatio = Number(progressCircle.style.strokeDashoffset) / Number(progressCircle.style.strokeDasharray)
+      const numCharacters = getPlainText(ensureActiveTab(mocks.getState()).feedbackState.editorState).length
+      const filledInRatio = numCharacters / 280
+      expect(remainingRatio).to.closeTo(0, 0.1)
```
Comment (will-weiss) [inline src/tests/integration/character-countdown.test.ts:61]: Make these `0.00000000000001` as 90% is not close enough to 100% for our liking.

File: src/tests/integration/character-countdown.test.ts [hunk 5]
```diff
@@ -0,0 +1,154 @@
...
+
+    it('has a .warning class', () => {
+      // TODO: Hank to write this test
+      const characterCountdown = mocks.app().querySelector('.character-countdown')! as HTMLDivElement
+      expect(characterCountdown.querySelectorAll('svg.warning')).to.have.length(1)
+    })
+
+    it('renders the number of characters remaining as a text element', () => {
+      // TODO: Hank to write this test
+      expect(mocks.app().querySelectorAll('.warning > text')).to.have.length(1)
```
Comment (will-weiss) [inline src/tests/integration/character-countdown.test.ts:73]: This should inspect the `.innerHTML` of the text element and assert its contents, which should be the number of characters remaining. 

```typescript
const text = mocks.app().querySelector('.character-countdown text')!
expect(text.innerHTML).to.equal('10')
```

File: src/tests/integration/character-countdown.test.ts [hunk 6]
```diff
@@ -0,0 +1,154 @@
...
+
+    it('has a .warning class', () => {
+      // TODO: Hank to write this test
+      const characterCountdown = mocks.app().querySelector('.character-countdown')! as HTMLDivElement
+      expect(characterCountdown.querySelectorAll('svg.warning')).to.have.length(1)
+    })
+
+    it('renders zero as the number of characters remaining in a text element', () => {
+      // TODO: Hank to write this test
+      const characterCountdown = mocks.app().querySelector('.character-countdown')! as HTMLDivElement
```
Comment (will-weiss) [inline src/tests/integration/character-countdown.test.ts:110]: Nice job! We can tidy this up a bit using the postfix `!` operator to tell typescript we know the text exists

```typescript
const text = mocks.app().querySelector('.character-countdown text')!
expect(text.innerHTML).to.equal('0')
```

**Review by will-weiss — CHANGES_REQUESTED** [review]

File: src/tests/integration/character-countdown.test.ts (hunk 2 above)
Comment (hankeliu2015) [reply src/tests/integration/character-countdown.test.ts:55]: done

File: src/tests/integration/character-countdown.test.ts (hunk 3 above)
Comment (hankeliu2015) [reply src/tests/integration/character-countdown.test.ts:9]: done

File: src/tests/integration/character-countdown.test.ts (hunk 6 above)
Comment (hankeliu2015) [reply src/tests/integration/character-countdown.test.ts:110]: resolved. thanks

File: src/tests/integration/character-countdown.test.ts (hunk 5 above)
Comment (hankeliu2015) [reply src/tests/integration/character-countdown.test.ts:73]: Thanks

File: src/popup/components/character-countdown.tsx (hunk 1 above)
Comment (hankeliu2015) [reply src/popup/components/character-countdown.tsx:24]: done

File: src/tests/integration/character-countdown.test.ts (hunk 4 above)
Comment (hankeliu2015) [reply src/tests/integration/character-countdown.test.ts]: syntax changed and meet the bar.

**Comment by hankeliu2015** [comment]
All change made as instructed. Please review. Thanks

**Review by will-weiss — APPROVED** [review]

**Pull request merged**